### ISSUE TITLE
[FR]: Support App Privacy Manifest

### ISSUE THREAD

**Issue by AlbGarciam** [firebase/firebase-ios-sdk#11490]
### Description

With the introduction of App Privacy manifests we're forced to declare which APIs are used, Domains where user tracking can be performed and the purpose per SDK.

https://developer.apple.com/videos/play/wwdc2023/10060/

Additionally any Firebase SDK that contains an app privacy manifest must be signed.

We would like to adopt this change as otherwise it should be added per app, and as we don't know which APIs/domains/tracking logic you're currently using we can't fulfil the privacy manifest properly

Thanks!

### Firebase Product(s)

Analytics, Remote Config

**Comment by paulb777** [comment]
Thanks for the request. We're actively investigating the implications of the WWDC 2023 Privacy Announcements.

In the meantime, https://firebase.google.com/docs/ios/app-store-data-collection is a reference for filling out the privacy manifest.

**Comment by AlbGarciam** [comment]
Just as a help, if you decide it is necessary to include the manifest as part of your libraries, you'll also need to sign the xcframeworks

**Comment by tboogh** [comment]
<img width="982" alt="image" src="https://github.com/firebase/firebase-ios-sdk/assets/582235/01022276-0e2d-45e7-b6d1-6342dcfd1ff3">

This should be clear enough that it's your responsability as a SDK provider to provide a privacy manifest.

**Comment by morganchen12** [comment]
We're working on providing SDK privacy manifests for Firebase. I don't have an ETA to share, but stay tuned.

**Comment by Shwethamugeraya** [comment]
@morganchen12  Please provide privacy manifests for Firebase.

**Comment by paulb777** [comment]
We're actively working on them and expect to deliver them in an upcoming release.

**Comment by cityfarmerNexon** [comment]
We are using both firebase analytics and firebase crashlytics.
When you are delivering, will it support both **static and dynamic framework**?
Also, do you have any set date for the release?

**Comment by paulb777** [comment]
Since Firebase currently only ships static frameworks in its binary distributions, we plan to first support that in the first few months of the new year.

We plan to add dynamic framework support when Firebase does its next major release, sometime later in the year.

**Comment by spdysgr** [comment]
Do you have any plans for which major SDK version will support Privacy Manifest? I'm using Firebase iOS SDK 8. Due to cost considerations, it is difficult for us to immediately upgrade to the latest major version. I know you are considering this, but I would like to ask you about your current plans.

**Comment by paulb777** [comment]
@spdysgr It will likely be done in a future minor version of major version 10, but if something unexpected comes up, could require a major version 11.

**Comment by mikehardy** [comment]
Just following along, but with respect to planning @spdysgr - if you are looking to plan + allocate resources, [the upstream documentation from Apple](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api) says this:

> From Spring 2024, apps that don’t describe their use of required reason API in their privacy manifest file won’t be accepted by App Store Connect

In terms of planning deadlines, the most pessimistic interpretation of "Spring 2024" is when it starts on March 19, 2024 (about 3 months from now) and the most optimistic interpretation is June 20, 2024 (about 6 months from now) before Apple begins enforcing this.

Last time I gambled on interpretations of vague dates with Apple they started enforcing something at the earliest / most pessimistic opportunity, which was unfortunate (for me). Something to keep in mind so that you're read for firebase-ios-sdk 10 (or maybe 11)

**Comment by spdysgr** [comment]
@paulb777 @mikehardy Thank you for your quick response. I understood that I need to upgrade to latest SDK version (currently 10, but could become 11 depending on the situation) as soon as possible. I will try planning to upgrade SDK in our team.

**Comment by LionWY** [comment]
+1

**Comment by sergiocampama** [comment]
FYI https://developer.apple.com/support/third-party-SDK-requirements/

**Comment by cityfarmerNexon** [comment]
> Since Firebase currently only ships static frameworks in its binary distributions, we plan to first support that in the first few months of the new year.
> 
> We plan to add dynamic framework support when Firebase does its next major release, sometime later in the year.

@paulb777 Thank you for your reply. I've got a quick question. According to the document below
https://developer.apple.com/support/third-party-SDK-requirements/

Firebase is included in the list which Apple announced for the commonly used SDK list.
and you said that you plan to the support in the first few months of the new year.

I wonder If newly supported static framework is xcframework or framework?
If It's xcframework, SDK signature will be supported for static framework also?

The document said that SDKs in the list must support SDK Signature.
so we want to know if this new policy should be applied to static framework also.

We are waiting for new framework to fulfill Apple's new requirement.

**Comment by ncooke3** [comment]
Hi @cityfarmerNexon, 

> I wonder If newly supported static framework is xcframework or framework?

_Static framework_ is equal to _xcframework_ in the context of Paul's [comment](https://github.com/firebase/firebase-ios-sdk/issues/11490#issuecomment-1834930465).

> If It's xcframework, SDK signature will be supported for static framework also?

Yes, correct, the xcframework would be signed.

**Comment by kalsariyarahul** [comment]
Please provide privacy manifests for Firebase.

**Comment by nokiaowner** [comment]
Hello,

Is there a tentative date and target release version for privacy manifest to be included in the Firebase SDK?

Thanks

**Comment by GH-Ong** [comment]
Hello,
am bumping for an ETA for privacymanifest and code signing requirements as well

**Comment by ncooke3** [comment]
Hi everyone, this is a top priority for the team and we are making progress. I do not have an ETA to share yet, but will keep this thread updated.

In the meantime, you can prepare by migrating to Firebase 10 if you're using an older major version.

If you're using Firebase 10, the eventual release containing manifests and SDK signatures will be a minor version release, so migration will consist of a package update and rebuild.

**Comment by alanzeino** [comment]
One thing that hasn't been mentioned so far, GoogleDataTransport is also on the list of frameworks Apple has outlined. 

Since that's an 'internal Google framework', only Google employees can PR a privacy manifest, yet one will be needed since it is a [dependency of this library](https://github.com/firebase/firebase-ios-sdk/blob/main/Package.swift#L152).

I guess Firebase's maintainers may be aware since [they've already commented on nanopb about this](https://github.com/nanopb/nanopb/issues/910), but worth mentioning in case anyone wonders why this is going to take some time. For teams taking this dependency from a different build system (i.e., Bazel), you would have to update all these libraries at the same time to meet Apple's requirements.

**Comment by wlxo0401** [comment]
+1

**Comment by lakshmankreditbee** [comment]
Hi, when can we expect an update regarding this issue?

**Comment by kalsariyarahul** [comment]
Hi, when can we expect an update regarding this issue?

**Comment by FilDevTronic** [comment]
@lakshmankreditbee @kalsariyarahul

https://github.com/firebase/firebase-ios-sdk/issues/11490#issuecomment-1881994104


> Hi everyone, this is a top priority for the team and we are making progress. I do not have an ETA to share yet, but will keep this thread updated.
> 
> In the meantime, you can prepare by migrating to Firebase 10 if you're using an older major version.
> 
> If you're using Firebase 10, the eventual release containing manifests and SDK signatures will be a minor version release, so migration will consist of a package update and rebuild.

**Comment by wlxo0401** [comment]
I discovered a new fact while testing. Until now, most of the libraries that have added PrivacyInfo have not been marked in PrivacyReport. 

However, when an item is added to the 'Personal Information Nutrition Label Type', the library name is displayed in the personal information protection report. 

Many libraries have updated PrivacyInfo without adding anything from Privacy Nutrition Label Types.

Does anyone know about this? Is "Privacy Nutrition Label Types" a must-have item?

Please feel free to leave your opinion.

**Comment by AlbGarciam** [comment]
> I discovered a new while testing.
> 
> Until now, most of the libraries that have added PrivacyInfo have not been marked in PrivacyReport.
> 
> However, when an item is added to the 'Privacy Nutrition Label Types', the library name is displayed in the PrivacyReport.
> 
> Many libraries have updated PrivacyInfo without adding anything from 'Privacy Nutrition Label Types'.
> 
> Does anyone know about this? Is 'Privacy Nutrition Label Types' a must-have item?
> 
> Or is it not necessary to display it in the privacy report?
> 
> Please feel free to leave opinion.
> 
> The 'PrivacyReport' I'm talking about can be downloaded as a PDF if you archive the app.
> 
> https://developer.apple.com/videos/play/wwdc2023/10060/ Please refer to 3:20.
> 
> 1. Open your project in Xcode.
> 2. Choose Product > Archive. Xcode creates the archive and reveals it in the organizer.
> 3. Control-click the archive in the organizer and choose Generate Privacy Report.
> 
> (I used a translator.)

The "Privacy Nutrition Label Types" are going to help you to declare the different cards you currently have to manually fill on the App Store once you want to publish a new app. By implementing the "Privacy Nutrition Label Types" within the xcprivacy file, Apple is going to automatically generate those cards and keep them up to date with your code. So yes it should be a must have in the libraries

**Comment by nokiaowner** [comment]
> > I discovered a new while testing.
> > Until now, most of the libraries that have added PrivacyInfo have not been marked in PrivacyReport.
> > However, when an item is added to the 'Privacy Nutrition Label Types', the library name is displayed in the PrivacyReport.
> > Many libraries have updated PrivacyInfo without adding anything from 'Privacy Nutrition Label Types'.
> > Does anyone know about this? Is 'Privacy Nutrition Label Types' a must-have item?
> > Or is it not necessary to display it in the privacy report?
> > Please feel free to leave opinion.
> > The 'PrivacyReport' I'm talking about can be downloaded as a PDF if you archive the app.
> > https://developer.apple.com/videos/play/wwdc2023/10060/ Please refer to 3:20.
> > 
> > 1. Open your project in Xcode.
> > 2. Choose Product > Archive. Xcode creates the archive and reveals it in the organizer.
> > 3. Control-click the archive in the organizer and choose Generate Privacy Report.
> > 
> > (I used a translator.)
> 
> The "Privacy Nutrition Label Types" are going to help you to declare the different cards you currently have to manually fill on the App Store once you want to publish a new app. By implementing the "Privacy Nutrition Label Types" within the xcprivacy file, Apple is going to automatically generate those cards and keep them up to date with your code. So yes it should be a must have in the libraries

@AlbGarciam Do you know where it's stated that Apple will automatically generate the Privacy nutrition cards on App Store based on the privacy manifest file? 

Correct me if I am wrong, It's our understanding (so far) that we will have to update those cards manually on app store based on the generated privacy report.

Thanks

**Comment by AlbGarciam** [comment]
> > > I discovered a new while testing.
> 
> > > Until now, most of the libraries that have added PrivacyInfo have not been marked in PrivacyReport.
> 
> > > However, when an item is added to the 'Privacy Nutrition Label Types', the library name is displayed in the PrivacyReport.
> 
> > > Many libraries have updated PrivacyInfo without adding anything from 'Privacy Nutrition Label Types'.
> 
> > > Does anyone know about this? Is 'Privacy Nutrition Label Types' a must-have item?
> 
> > > Or is it not necessary to display it in the privacy report?
> 
> > > Please feel free to leave opinion.
> 
> > > The 'PrivacyReport' I'm talking about can be downloaded as a PDF if you archive the app.
> 
> > > https://developer.apple.com/videos/play/wwdc2023/10060/ Please refer to 3:20.
> 
> > > 
> 
> > > 1. Open your project in Xcode.
> 
> > > 2. Choose Product > Archive. Xcode creates the archive and reveals it in the organizer.
> 
> > > 3. Control-click the archive in the organizer and choose Generate Privacy Report.
> 
> > > 
> 
> > > (I used a translator.)
> 
> > 
> 
> > The "Privacy Nutrition Label Types" are going to help you to declare the different cards you currently have to manually fill on the App Store once you want to publish a new app. By implementing the "Privacy Nutrition Label Types" within the xcprivacy file, Apple is going to automatically generate those cards and keep them up to date with your code. So yes it should be a must have in the libraries
> 
> 
> 
> @AlbGarciam Do you know where it's stated that Apple will automatically generate the Privacy nutrition cards on App Store based on the privacy manifest file? 
> 
> 
> 
> Correct me if I am wrong, It's our understanding (so far) that we will have to update those cards manually on app store based on the generated privacy report.
> 
> 
> 
> Thanks

As far as they told me during WWDC yes

**Comment by omarzl** [comment]
You could use the generated pdf as a guide but the privacy nutrition labels on App Store Connect MUST be updated manually. I confirmed this with Apple folks.

**Comment by StartingZhou** [comment]
New release version of Firebase, still no privacy manifest, could you please help check?

**Comment by paulb777** [comment]
We weren't able to complete the privacy manifests in time for 10.21.0 and are hopeful we can release in 10.22.0, due out the week of February 26th.

**Comment by hongchanglin** [comment]
> We weren't able to complete the privacy manifests in time for 10.21.0 and are hopeful we can release in 10.22.0, due out the week of February 26th.

Thank you for your continued support and the timely update regarding the privacy manifest situation.

However, as far as I know, Google Firebase provides multiple SDKs. May I confirm if all of these SDKs will include privacy manifest files in accordance with Apple's regulations?

**Comment by paulb777** [comment]
We're working on privacy manifests for all Firebase SDKs and dependencies.

**Comment by JLLA113** [comment]
> We're working on privacy manifests for all Firebase SDKs and dependencies.

Hi,

In that version, Will you support the sign for sdk frameworks? 

Regards!

**Comment by paulb777** [comment]
Signing will take longer and may not make the same release as the manifests, but it will follow as soon as we can. Signing is tracked at #12238.

**Comment by sylvia-tencent-wmt** [comment]
> Privacy manifests for the Firebase SDKs named in https://developer.apple.com/support/third-party-SDK-requirements/ have merged to `main` in #12407 and is planned to be part of the next release due out the week of March 5th.

Does the GoogleSignIn SDK support Privacy Manifests?

**Comment by clausjoergensen** [comment]
> Privacy manifests for the Firebase SDKs named in https://developer.apple.com/support/third-party-SDK-requirements/ have merged to `main` in #12407 and is planned to be part of the next release due out the week of March 5th.

@paulb777 you'll need to update GoogleAppMeasurement as well, as it's currently pinning `nanopb` (one of the SDKs named on that list) to the version before the privacy manifest was introduced.

See https://github.com/google/GoogleAppMeasurement/blob/main/Package.swift#L43-L47

**Comment by clausjoergensen** [comment]
> Privacy manifests for the Firebase SDKs named in https://developer.apple.com/support/third-party-SDK-requirements/ have merged to `main` in #12407 and is planned to be part of the next release due out the week of March 5th.

You'll need to update GoogleAppMeasurement as well, as it's currently pinning `nanopb` (one of the SDKs named on that list) to the version before the privacy manifest was introduced.

See https://github.com/google/GoogleAppMeasurement/blob/main/Package.swift#L43-L47

**Comment by ncooke3** [comment]
Hi @sylvia-tencent-wmt,

> Does the support for Privacy Manifests include the GoogleSignIn SDK?

The GoogleSignIn SDK has a different release schedule, so GoogleSignIn SDK's support of privacy manifests may not exactly align with the upcoming Firebase release. I would continue to follow https://github.com/google/GoogleSignIn-iOS/issues/341 for updates.

**Comment by GH-Ong** [comment]
Hi is it possible to get a date for release with privacyManifest and code signing signatures as well?

**Comment by paulb777** [comment]
See https://github.com/firebase/firebase-ios-sdk/issues/11490#issuecomment-1962119838 for named Firebase privacy manifests. We're working on code signing, but don't have a date yet. Follow #12238 for updates on signing.

**Comment by andrewheard** [comment]
The [10.22.0 release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.0) added privacy manifests for the Firebase SDKs named in https://developer.apple.com/support/third-party-SDK-requirements/. Signing support will continue to be tracked in https://github.com/firebase/firebase-ios-sdk/issues/12238.

**Comment by ManoharaReddyP** [comment]
Hi @andrewheard, 

I have added FirebaseABTesting.xcframework to the project and generate privacy report I am still getting error like "The archive does not contain any `PrivacyInfo.xcprivacy` files"  
Any solution for this? Also I noticed you guys have added "FirebaseABTesting_Privacy.bundle" but when I archive it's not added in the final bundle of the app. I am using Xcode 15.

Any recommendations?
